Automattic/woocommerce-payments
# Complete an order if payment intent is successful

Fixes https://github.com/Automattic/woocommerce-payments/issues/1565

#### Changes proposed in this Pull Request
This PR accepts a new event `payment_intent.succeeded` from the server. When the event `payment_intent.succeeded` is received, this PR will look up the order for this payment and update its status to "Processing"/"Completed". 

Changes:
- Accepts `payment_intent.succeeded` event in `class-wc-rest-payments-webhook-controller.php`
- Use `WC_Order`'s [payment_complete](https://github.com/woocommerce/woocommerce/blob/e4fec5b914d523ca7312506bd07f86ccc5cb30d5/includes/class-wc-order.php#L99) method to complete the order when payment is processed.
- Refactor `class-wc-payments-db.php` to retrieve payment data by `_intent_id` or `_charge_id`.

#### Testing instructions
Similar to the setup in https://github.com/Automattic/woocommerce-payments/pull/1537
1. Set store currency to euro
2. Enable feature flag wp option, update `_wcpay_feature_sepa` to `1`
3. Enable SEPA in checkbox at bottom of `/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments`
4. Add item to cart and go to checkout
5. Go to Orders page `wp-admin/edit.php?post_type=shop_order` and you should see the payment in "On Hold" status.
6. Go to your local wcpay server and manually trigger a webhook event via cli command. Please feel free to ping me if you need the command/details on this.
7. Go to Orders page again. Depending on if the order require fulfilment, its status should be updated to either "Processing" or "Completed".

-------------------
